dev/ci/gitlab-ci-ctests.yml

```diff
@@ -45,9 +53,49 @@
   extends: .base_config
   stage: run_tests
   script: |
-    set -e  # Fail the job if any command fails
-    echo "Running ${CTEST} tests in ${GW_HOMEgfs}/dev/ctests"
+    echo "Using build directory ${GW_HOMEgfs}"
+    source ${GW_HOMEgfs}/dev/ush/gw_setup.sh
     cd ${GW_HOMEgfs}/dev/ctests/build
-    ctest -L ${CTEST} --output-on-failure
+    echo "Running CTests with label: ${CTEST_NAME} in ${PWD}"
+    num=$(ctest -N | grep "Total Tests" | awk "{print \$3}")
+    [ "$num" -gt 0 ] || { echo "No tests found"; exit 1; }
```

Comment:
Braces around `num`.  Also, you don't need to quote numerical variables when using `-gt`, `-le`, etc.  Lastly, please make this an if block.